povm_toolbox/post_processor/median_of_means.py

```diff
@@ -0,0 +1,196 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""MedianOfMeans."""
+
+from __future__ import annotations
+
+import sys
+
+if sys.version_info < (3, 12):
+    from typing_extensions import override
+else:
+    from typing import override  # pragma: no cover
+
+
+import numpy as np
+from numpy.random import Generator, default_rng
+from qiskit.quantum_info import SparsePauliOp
+
+from povm_toolbox.post_processor.povm_post_processor import POVMPostProcessor
+from povm_toolbox.quantum_info.base import BaseDual
+from povm_toolbox.sampler import POVMPubResult
+
+
+class MedianOfMeans(POVMPostProcessor):
+    """A POVM result post-processor which uses a 'median of means' estimator.
```

Comment:
Can you explain in this docstring a bit how this differs from the base class?
Points to mention would be:
- maybe an equation for the median of means approach
- its benefits
- its drawbacks (?)